src/lib/PerceptronPredictor.cc

```diff
@@ -143,6 +156,19 @@ void PerceptronPredictor::flush(uint64_t address) {
     }
     rasHistory_.erase(it);
   }
+
+  // If possible, pop instruction from FTQ
+  // if (!FTQ_.empty()) FTQ_.pop_back();
+  FTQ_.pop_back();
+
+  // Roll back global history
+  globalHistory_ >>= 1;
+}
```

Comment:
If a flush occurs, do we know it will always be because of the most recently processed branch?

Reply:
See my reply to your comment re GenericPredictor.cc line 131